src/target/sfdp.c

```diff
@@ -82,7 +83,14 @@ static spi_parameters_s sfdp_read_basic_parameter_table(
 			break;
 		}
 	}
-	result.page_size = SFDP_PAGE_SIZE(parameter_table);
+	// The timing and page size DWORD was added in JESD216A. Before then, the
+	// table is 9 DWORDs and we default to page size 256.
+	if (length >= 4 * 11) {
```

Comment:
This if-else block has excess braces, please remove them. As mentioned in Discord we'd prefer to use a version check than a length check here because Vendors have a habit of doing silly things that screw these kinds of check up. We'll add a suggestion for how to do that from the SFDP spec when we get a chance to properly review it.

Reply:
I'll update based on version number when you have a suggestion. It wasn't clear to me when I looked because I couldn't find the old spec. The changes described in the current doc doesn't make it super clear what the old spec said for versioning.

Reply:
Reading the standard's page 27 ("Page 13") - §6.2.2, it shows that JESD26F uses version major 1, version minor 10; JESD26E uses version major 1 and minor 9, JESD25D major 1 and minor 8 and then before that, in JESD25C it was major 0 and an unknown minor number. Your part has major 0 and minor 1:
```
8 byte SFDP read at 0x0:
        53 46 44 50 00 01 01 ff
```
so it's reasonable to assume this is an original standard part or one following the second release.

There are another pair of version numbers in the table header following a similar numbering, and with your part having:
```
8 byte SFDP read at 0x8:
        00 00 01 09 30 00 00 ff
```
which defines the table major to 0 and minor to 1 while the current standard is 1 and 8.

If this is passed into this function as a `const sfdp_parameter_table_header_s *const header` parameter, we can test `if (header->version_major > 1 || (header->version_major == 1 && header->version_minor >= 5))`.

These values are derived from Annex B's (sketchy..) data on what changed in each JESD216 letter suffixed standard, and in A -> B they specify this header goes to minor 6 by incrementing the number, with the no letter -> A change saying the main header gets major 1 and minor 5. It's in this no-to-A change that the parameter table is changed from 9 to 16 uint32_t's and the value we're after here is in the 11th.